### ISSUE TITLE
mAP using this seems low

### ISSUE THREAD

**Issue by JNatael** [experiencor/basic-yolo-keras#138]
It seems like the mAP values on this are a bit low, if I'm understanding them correctly. The mAP at https://github.com/experiencor/basic-yolo-keras says you get a max value of just over 42, while the original YOLOv2 was closer to 70 I believe. Do you have thoughts on why it's so much lower? Know of other users who have managed to do better?

I'm in a group using an implementation of this in a project and currently our implementation has topped out at around 48 mAP (also using the cocoapi to calculate it) and so far we've been unable to push it higher. Normally I'd start to wonder if there was an error in the loss function to get values so much below the published paper results on a pre-trained network, but we've found no error so far and if others are using this with success then that seems far less likely too. Which leaves me somewhat at a loss though; thoughts on what accounts for the difference? Or am I misunderstanding something? Or are others able to close this gap and these two instances just don't?

Many thanks for sharing a great project.

**Comment by experiencor** [comment]
The best result on COCO is 42 and on PASCAL is 76. May I know whether your score of 48 is on COCO or PASCAL?

FYI, I have yet to implement multi-scale training. It may account for the difference in the score.

**Comment by JNatael** [comment]
It was on a very small subset of COCO.

**Comment by JNatael** [comment]
Is the COCO result you mention just from your experience or is that generally known somewhere? And if the latter, can you point me to that source? Would like to dig into that more to understand.

**Comment by experiencor** [comment]
@JNatael Very sorry for the delay. I was on holiday. The one that scores 42 is the official YOLO weight (https://pjreddie.com/media/files/yolo-voc.weights). You may want to refer to Yolo Step by Step for loading this weight to the FullYolo network.

**Comment by algila** [comment]
Hi, may you confirm if my understanding is correct.
running your code train.py using Full yolo (pretrained weights downloaded from your provided link) will be obtained mAp = 28,6
running your code yolo step by step using yolo weight from pjreddie web page will be obtained mAp = 42

Is the different mAP due to only the file with the weights used ?

**Comment by experiencor** [comment]
@algila Running train.py will get you to mAP 28.6. The mAP 42 is scored by the official YOLO model (https://pjreddie.com/media/files/yolo-voc.weights).

**Comment by algila** [comment]
In Yolo Step-by-Step.ipynb you upload the weights from the official YOLO mode (In [7] and In [8]) so I suppose you should catch mAP=42 but in the following line (In [13]) why you overwrite the official weights reinizializing last Conv Layer with random ?
```
layer   = model.layers[-4] # the last convolutional layer
weights = layer.get_weights()

new_kernel = np.random.normal(size=weights[0].shape)/(GRID_H*GRID_W)
new_bias   = np.random.normal(size=weights[1].shape)/(GRID_H*GRID_W)

layer.set_weights([new_kernel, new_bias])
```
Is it correct that just commenting the above we should obtain mAP=42 ?

**Comment by experiencor** [comment]
@algila Yes, you are right. I randomize the last layer just to train the network again as a test to see if my custom loss is implemented correctly.
